src/StoreApi/Authentication.php

```diff
@@ -20,15 +22,51 @@ public function init() {
 	 * @return \WP_Error|null|bool
 	 */
 	public function check_authentication( $result ) {
-		if ( ! empty( $result ) ) {
+		if ( ! $this->is_request_to_store_api() ) {
 			return $result;
 		}
 
-		if ( $this->is_request_to_store_api() ) {
-			return true;
+		/**
+		 * Filters the Store API rate limit check.
+		 *
+		 * This can be used to disable the rate limit check when testing API endpoints via a REST API client.
+		 *
+		 * @param boolean $disable_rate_limit_check If true, checks will be disabled.
+		 * @return boolean
+		 */
+		if ( ! apply_filters( 'woocommerce_store_api_disable_rate_limit_check', false ) ) {
```

Comment:
Switch this to off by default.

```suggestion
		if ( apply_filters( 'woocommerce_store_api_enable_rate_limit_check', false ) ) {
```